src/radical/saga/adaptors/lsf/lsfjob.py

```diff
@@ -234,6 +234,8 @@ def _lsfscript_generator(url, logger, jd, ppn, lsf_version, queue):
     if smt not in SMT_VALID_VALUES:
         smt = SMT_DEFAULT
 
+    assert (int(jd.environment.get('RADICAL_SMT', 1)) == smt)
```

Comment:
I am not sure about the assert.  For example, `smt` might not be set in the system arch.  Should we really assert if `RADICAL_SMT` is different than the default value in this case - shouldn't we just use the env value?

Reply:
Didn't want to have them out of sync between each other. But I agree to have env variable with higher priority, and then if it is not set, then check SMT from `system_architecture` (and remove assert)

p.s. and it would be truly in sync with RP, since RP checks env variable first

committed accordingly